ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -131,6 +127,15 @@ constructor(
     }
   }
 
+  suspend fun activateSurveyInternal(surveyId: String): Survey {
```

Comment:
I actually have a big PR pending that updates the loading state here explicitly, making it a little easier to follow. I'll apply that change once I get to merging things properly.

Reply:
Thanks for the heads up! I've reverted changes to the survey repository to unblock this PR and delegated the survey loading/activation flow to HomeScreenFragment.